src/platform/plugins/shared/data_views/public/content_management_wrapper.ts

```diff
@@ -58,7 +58,7 @@ export class ContentMagementWrapper implements PersistenceAPI {
       });
     } catch (e) {
       if (e.body?.statusCode === 404) {
-        throw new SavedObjectNotFound('data view', id, 'management/kibana/dataViews');
```

Comment:
This change and the changes to `SavedObjectNotFound` have a pretty broad impact since they affect the display of the error in all areas of Kibana, not just alerting rules. Instead, could there just be special handling and messaging for the error within alerting rules where the link doesn't make sense? The relevant properties for detecting `savedObjectType` and `savedObjectId` already exist on `SavedObjectNotFound` anyway, and it would retain the link for areas where it makes sense (e.g. Discover).

Reply:
From the linked issue, it looks like the error message is being rendered as plain text for rules, but it's actually markdown and is rendered correctly in the toast shown in Discover, for example. Tbh there are probably better ways to handle this like adding a separate `savedObjectLink` property to the error, removing the message markdown, and rendering it separately in the toast, but that might be more effort than it's worth for what you're trying to fix here.